### PR TITLE
Update Travis CI config

### PR DESCRIPTION
**Pull request by webimpress**
- removes hhvm build
- adds php 7.1, 7.2 ~~and nightly (7.3-dev) builds~~ (removed as not even alpha has been published)
- removes irc notifications
- adds composer.lock to the repository and run tests with lowest/locked/latest dependencies

### CODE REVIEW

**Comment by webimpress** [comment]
Any ideas why the last build fails? Is it the issue with cache? Maybe we should add also `vendor` in travis ci `cache.directories`?

**Comment by kokspflanze** [comment]
@webimpress did you mean 

``
Parse error: syntax error, unexpected ':', expecting ';' or '{' in /home/travis/build/zendframework/zend-db/vendor/doctrine/instantiator/src/Doctrine/Instantiator/Instantiator.php on line 95
``

**Comment by webimpress** [comment]
@kokspflanze oh, you have right, thanks! The travis configuration here is so different (than others libraries) and probably we can improve it a lot... Definitely we need to add PHP 7.2 support in the library... I need to have another look on it.

File: .travis.yml [hunk 1]
```diff
@@ -10,88 +10,91 @@ branches:
...
-        - DEPLOY_DOCS="$(if [[ $TRAVIS_BRANCH == 'master' && $TRAVIS_PULL_REQUEST == 'false' ]]; then echo -n 'true' ; else echo -n 'false' ; fi)"
-        - PATH="$HOME/.local/bin:$PATH"
+        - DEPS=lowest
     - php: 5.6
       env:
-        - EVENT_MANAGER_VERSION='^2.6.2'
-        - HYDRATOR_VERSION='^1.1'
-        - SERVICE_MANAGER_VERSION='^2.7.5'
-    - php: 7
-    - php: 7
```
Comment (kokspflanze) [inline .travis.yml]: dont miss to readd php 7 in the matrix

Comment (webimpress) [inline .travis.yml]: Oh, good point, I've missed that. Thanks. Done in c91b08b17a769ec714dbd15abb476f19793f2807

**Comment by webimpress** [comment]
Finally green !

/cc @ezimuel @weierophinney @mwillbanks

**Comment by akrabat** [comment]
I"m getting: 

```
$ vendor/bin/phpunit -d xdebug.scream=0
PHPUnit 4.8.36 by Sebastian Bergmann and contributors.

.....................SSSSSSSSSSSS...SS.....IIIIIIIIIIIIIIIIII   61 / 1092 (  5%)
SSSS...IIIIIISSSSSSSSSSSS.......SS.....IIIIIIIIIIIIIIIIII....  122 / 1092 ( 11%)
...SSSS.......I.I.I......III..I..............................  183 / 1092 ( 16%)
..........S..SSEE......IIIIISSSSSSSSSSSSS...IIIIIIIIIIIIISS..  244 / 1092 ( 22%)
...IIIIISSSS...IIIIII................................SS......  305 / 1092 ( 27%)
.....................SS......................S...............  366 / 1092 ( 33%)
.......................S.....................................  427 / 1092 ( 39%)
.............................................................  488 / 1092 ( 44%)
.............................S...............................  549 / 1092 ( 50%)
.............................................................  610 / 1092 ( 55%)
.............................................................  671 / 1092 ( 61%)
.............................................................  732 / 1092 ( 67%)
.............................................................  793 / 1092 ( 72%)
.............................................................  854 / 1092 ( 78%)
.............................................................  915 / 1092 ( 83%)
.............................................................  976 / 1092 ( 89%)
............................................................. 1037 / 1092 ( 94%)
.......................................................

Time: 2.21 seconds, Memory: 24.00MB

There were 2 errors:

1) ZendTest\Db\Adapter\Driver\Pgsql\ConnectionTest::testSetCharset
PHPUnit_Framework_Exception: PHP Deprecated:  ini_set(): Use of iconv.input_encoding is deprecated in - on line 70
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:70
PHP Deprecated:  ini_set(): Use of iconv.internal_encoding is deprecated in - on line 71
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:71
PHP Deprecated:  ini_set(): Use of iconv.output_encoding is deprecated in - on line 72
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:72
PHP Deprecated:  ini_set(): Use of mbstring.http_output is deprecated in - on line 98
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:98
PHP Deprecated:  ini_set(): Use of mbstring.internal_encoding is deprecated in - on line 100
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:100

2) ZendTest\Db\Adapter\Driver\Pgsql\ConnectionTest::testSetInvalidCharset
PHPUnit_Framework_Exception: PHP Deprecated:  ini_set(): Use of iconv.input_encoding is deprecated in - on line 70
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:70
PHP Deprecated:  ini_set(): Use of iconv.internal_encoding is deprecated in - on line 71
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:71
PHP Deprecated:  ini_set(): Use of iconv.output_encoding is deprecated in - on line 72
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:72
PHP Deprecated:  ini_set(): Use of mbstring.http_output is deprecated in - on line 98
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:98
PHP Deprecated:  ini_set(): Use of mbstring.internal_encoding is deprecated in - on line 100
PHP Stack trace:
PHP   1. {main}() -:0
PHP   2. ini_set() -:100

FAILURES!
Tests: 1092, Assertions: 1759, Errors: 2, Skipped: 65, Incomplete: 78.

**Pull request merged**